packages/walletconnect/src/index.ts

```diff
@@ -75,47 +69,37 @@ export class WalletConnect extends Connector {
     if (error) this.onError?.(error)
   }
 
-  private chainChangedListener = (chainId: number | string): void => {
-    this.actions.update({ chainId: parseChainId(chainId) })
+  private chainChangedListener = (chainId: number): void => {
+    this.actions.update({ chainId })
   }
 
   private accountsChangedListener = (accounts: string[]): void => {
     this.actions.update({ accounts })
   }
 
-  private URIListener = (_: Error | null, payload: { params: string[] }): void => {
-    this.events.emit(URI_AVAILABLE, payload.params[0])
+  private URIListener = (uri: string): void => {
+    this.events.emit(URI_AVAILABLE, uri)
   }
 
-  private async isomorphicInitialize(chainId = this.defaultChainId): Promise<void> {
-    if (this.eagerConnection) return
-
-    // because we can only use 1 url per chainId, we need to decide between multiple, where necessary
-    const rpc = Promise.all(
-      Object.keys(this.rpc).map(
-        async (chainId): Promise<[number, string]> => [
-          Number(chainId),
-          await getBestUrl(this.rpc[Number(chainId)], this.timeout),
-        ]
-      )
-    ).then((results) =>
-      results.reduce<{ [chainId: number]: string }>((accumulator, [chainId, url]) => {
-        accumulator[chainId] = url
-        return accumulator
-      }, {})
-    )
-
-    return (this.eagerConnection = import('@walletconnect/ethereum-provider').then(async (m) => {
-      this.provider = new m.default({
+  private async isomorphicInitialize(
+    desiredChainId: number | undefined = this.defaultChainId
+  ): Promise<MockWalletConnectProvider> {
+    if (this.eagerConnection) return this.eagerConnection
+
+    const chains = desiredChainId ? orderToSetDefaultChain(this.chains, desiredChainId) : this.chains
+    const rpcMap = this.rpcMap ? await getRpcBestUrlMap(this.rpcMap, this.timeout) : undefined
```

Comment:
This awaited Promise should be captured by this.eagerConnection so that subsequent initializations are isomorphic, otherwise it's possible to re-initialize while waiting for the urls to resolve. Could you restructure to put this in the thennable after L92, or just omit the `await` and `await` it inside of the thennable?

You may want to add a test against this too to make sure the code doesn't regress in the future.